app/controllers/provider_interface/offers_controller.rb

```diff
@@ -87,22 +85,32 @@ def action
       'back' if !!params[:back]
     end
 
+    def query_service
+      @query_service ||= GetChangeOfferOptions.new(
+        user: current_provider_user,
+        current_course: @application_choice.offered_course,
+      )
+    end
+
```

Comment:
Can we move this further down so the methods accessed by the controller's public methods are more prominent in the controller.